Assignment 2.1: Continuous Integration in GitHub

ManishPanda-ta opened the pull request:
## Description

This pull request addresses the tasks outlined in **Issue #7 :** Complete Assignment 2.1: Continuous Integration in GitHub. The changes include:

- Setting up an automated workflow using GitHub Actions to create a Conda environment.
- Running flake8 tests as part of the CI process.
- Executing nonstandardcode.py as part of the CI process.
- Adding necessary documentation and comments for clarity.
- Ensuring all tests pass and the code adheres to the project's style guidelines.

## Linked Issue

Closes #7 

## Checklist

- [x] Code follows the project's style guidelines
- [x] Documentation has been updated
- [x] All tests have been run and pass
- [x] Screenshots of the workflow run outputs have been added to the PR description

## Deliverables
1. Output of flake8 in the workflow run.
2. Output of nonstandardcode.py in the workflow run.
3. Screenshots of the above outputs in the PR description.

Link to Deliverables Screenshots : 
https://docs.google.com/document/d/1Q5MAi-wU2anZyJoeAVaXb4DEMCzyRCzRFwY0sh2Mv3E/edit?usp=sharing

---
Please review the changes and let me know if there are any issues or further adjustments needed.

RakshitRK3352 commented:
@ManishPanda-ta , Do not git track datasets, Please add them to gitignore file.

ManishPanda-ta commented:
> @ManishPanda-ta , Do not git track datasets, Please add them to gitignore file.

Hi @RakshitRK3352 , as instructed I have removed datasets from the remote repos and added them to .gitignore .
